### PR TITLE
feat(gate)!:  wasmtime support

### PR DESCRIPTION
**Pull request by afmika**
#### Motivation and context

Enable support for [wit](https://github.com/WebAssembly/component-model/blob/main/design/mvp/WIT.md) and facilitate readiness for the upcoming specs.

#### Migration notes
`#[wasmedge_bindgen]` are replaced by wit bindings.

In the old version we were restricted to`#[wasmedge_bindgen]`, which was only available in Rust and unique to WasmEdge.
```rust
#[wasmedge_bindgen]
fn add(a: u32, b: u32) -> u32 {
   a + b
}
```

In the new implementation,  wasm modules written in any language that uses the `wit` interface are now natively supported within `typegate`.
```wit
// wit/example.wit
package example:host;
world host {
  export add: func(a: u32, b: u32) -> u32;
}
```
An implementation (eg. in Rust) may look like this..
```rust
// src/lib.rs
wit_bindgen::generate!({ world: "host" });
struct MyLib;
impl Guest for MyLib {
    fn add(a: u32, b: u32) -> u32 {
        a + b
    }
}
export!(MyLib);
```

### Checklist

- [x] The change come with new or modified tests
- [x] Hard-to-understand functions have explanatory comments
- [ ] End-user documentation is updated to reflect the change

### CODE REVIEW

File: typegate/tests/runtimes/wasm/wasm_test.ts
Comment (zifeo) [inline typegate/tests/runtimes/wasm/wasm_test.ts:13]: deadcode?

**Review by zifeo — COMMENTED** [review]
Can you add an example in the migration description?

File: typegate/tests/runtimes/wasm/wasm_test.ts
Comment (afmika) [reply typegate/tests/runtimes/wasm/wasm_test.ts:13]: @destifo I guess once you find a workaround this can be uncommented.

Comment (destifo) [reply typegate/tests/runtimes/wasm/wasm_test.ts:13]: Yeah, I am working on it now.

**Review by zifeo — COMMENTED** [review]
lgtm, waiting on another review with that large PR :)

**Review by Yohe-Am — APPROVED** [review]

**Review by destifo — APPROVED** [review]

**Pull request merged**
